app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,25 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).first
+    if user
+      @token = user.token
+    else
+      redirect_to expired_token_path
+    end
+  end
+
+  def expired_token; end
+
+  def create
+    user = User.where(token: params[:token]).first
+    if user 
+      user.password = params[:password]
+      user.generate_token
+      user.save
```

Comment:
I'm no longer generating token here, I'm doing it with the request to reset the password is made, before the actual email goes out. Much better this way.